File: services/user-tenant-service/src/main/java/com/sourcefuse/jarc/services/usertenantservice/dto/UserDto.java
package com.sourcefuse.jarc.services.usertenantservice.dto;

import com.sourcefuse.jarc.core.enums.UserStatus;
import jakarta.validation.Valid;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class UserDto {

  @NotNull(message = "Role ID cannot be null")
  private UUID roleId;

  @NotNull(message = "Tenant ID cannot be null")
  private UUID tenantId;

  private UserStatus status;

  private String authProvider;

  private String authId;

  private UUID userTenantId;

  @Valid
  @NotNull(message = "User details cannot be null")
  private User userDetails;

  public UserDto(
    User existingUser,
    UUID roleId,
    UserStatus status,
    UUID tenantId,
    UUID userTenantId,
    String authProvider
  ) {
    this.userDetails = existingUser;
    this.roleId = roleId;
    this.status = status;
    this.tenantId = tenantId;
    this.userTenantId = userTenantId;
    this.authProvider = authProvider;
  }
}

File: examples/basic-example/frontend/example-app/src/app/shared/DTO/user.dto.ts
import { User } from 'src/app/core/models';

export class UserDto {
  authProvider: string;
  userTenantId: string;
  authId: string;
  tenantId: string;
  roleId: string;
  status: string;
  userDetails: User;

  constructor(data?: Partial<UserDto>) {
    this.authProvider = data?.authProvider;
    this.userTenantId = data?.userTenantId;
    this.authId = data?.authId;
    this.tenantId = data?.tenantId;
    this.roleId = data?.roleId;
    this.status = data?.status;
    this.userDetails = new User(data?.userDetails);
  }
}
